### PR TITLE
Unittest2pytest store test file store

### PR DESCRIPTION
**Pull request by wkrt7**
Signed-off-by: Wiktor Wasowicz [799637+wkrt7@users.noreply.github.com](mailto:799637+wkrt7@users.noreply.github.com)

## Related Issues/PRs

https://github.com/mlflow/mlflow/issues/7201

 #xxx

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/store/model_registry/test_file_store.py [hunk 1]
```diff
@@ -1,1364 +1,1341 @@
...
-                other_model_name = name
-                break
-        with pytest.raises(
-            MlflowException, match=rf"Registered Model \(name={other_model_name}\) already exists."
-        ):
-            fs.rename_registered_model(model_name, other_model_name)
+@pytest.fixture(scope="function")
+def test_root(tmp_path_factory):
+    tmp_path = str(tmp_path_factory.mktemp("test_root") / f"test_file_store_{random_int()}")
+    return tmp_path
```
Comment (harupy) [inline tests/store/model_registry/test_file_store.py:31]: Do we need this fixture? Can we use `tmp_path` fixture?

File: tests/store/model_registry/test_file_store.py [hunk 2]
```diff
@@ -1,1364 +1,1341 @@
...
 
-        new_name = model_name + "!!!"
-        assert model_name != new_name
-        fs.rename_registered_model(model_name, new_name)
-        assert fs.get_registered_model(new_name).name == new_name
 
-    def _extract_names(self, registered_models):
-        return [rm.name for rm in registered_models]
+@pytest.fixture(scope="function")
```
Comment (BenWilson2) [inline tests/store/model_registry/test_file_store.py:34]: Do we need to explicitly declare that the scope is the default value?

File: tests/store/model_registry/test_file_store.py [hunk 3]
```diff
@@ -1,1364 +1,1341 @@
...
-            MlflowException,
-            match=rf"Model Version \(name={mv.name}, version={mv.version}\) not found",
-        ) as exception_context:
-            fs.update_model_version(mv.name, mv.version, description="deleted!")
-        assert exception_context.value.error_code == ErrorCode.Name(RESOURCE_DOES_NOT_EXIST)
+        _verify_registered_model(store, name, rm_data)
 
-        # cannot delete it again
+    # test that fake registered models dont exist.
+    for name in {random_str(25) for _ in range(10)}:
```
Comment (harupy) [inline tests/store/model_registry/test_file_store.py:98]: Do we need this loop?

File: tests/store/model_registry/test_file_store.py [hunk 4]
```diff
@@ -1,1364 +1,1341 @@
...
-            ),
-        ) as exception_context:
-            search_versions("run_id IN (1,2,3)")
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
-        assert set(search_versions(f"run_id LIKE '{run_id_2[:30]}%'")) == {2, 3}
+def test_rename_registered_model(store, registered_models_names, rm_data):
+    model_name = registered_models_names[random_int(0, len(registered_models_names) - 1)]
+    # Error cases
+    with pytest.raises(MlflowException, match="Registered model name cannot be empty."):
```
Comment (harupy) [inline tests/store/model_registry/test_file_store.py:115]:
```suggestion
    with pytest.raises(MlflowException, match=r"Registered model name cannot be empty\."):
```
```suggestion
    with pytest.raises(MlflowException, match="Registered model name cannot be empty."):
```

File: tests/store/model_registry/test_file_store.py [hunk 5]
```diff
@@ -1,1364 +1,1341 @@
...
+        store.rename_registered_model(model_name, None)
+    # test that names of existing registered models are checked before renaming
 
-        assert set(search_versions(f"run_id ILIKE '{run_id_2[:30].upper()}%'")) == {2, 3}
+    for name in registered_models_names:
```
Comment (BenWilson2) [inline tests/store/model_registry/test_file_store.py:119]: Do we need to do this? Why the random selection here? Why not just take the last and first entries from registered_model_names and assert that we can't rename the last as the first?

File: tests/store/model_registry/test_file_store.py [hunk 6]
```diff
@@ -1,1364 +1,1341 @@
...
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
-
-        # search using an ill-formed IN operator correctly throws exception
-        with pytest.raises(
-            MlflowException, match=r"Invalid clause\(s\) in filter string"
-        ) as exception_context:
-            search_versions("run_id IN (")
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+    new_name = model_name + "!!!"
+    assert model_name != new_name
```
Comment (BenWilson2) [inline tests/store/model_registry/test_file_store.py:129]: This will never throw.

File: tests/store/model_registry/test_file_store.py [hunk 7]
```diff
@@ -1,1364 +1,1341 @@
...
-        ) as exception_context:
-            search_versions("run_id IN")
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
-        with pytest.raises(
-            MlflowException, match=r"Invalid clause\(s\) in filter string"
-        ) as exception_context:
-            search_versions("name LIKE")
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+def _extract_names(registered_models):
```
Comment (BenWilson2) [inline tests/store/model_registry/test_file_store.py:134]: this utility function can go above the first test

File: tests/store/model_registry/test_file_store.py [hunk 8]
```diff
@@ -1,1364 +1,1341 @@
...
-
-        assert set(search_versions(None)) == {1, 2, 3}
-
-        assert set(search_versions("name='%s'" % name)) == {1, 2, 3}
-
-        fs.transition_model_version_stage(
-            name=mv1.name, version=mv1.version, stage="production", archive_existing_versions=False
-        )
+def test_delete_registered_model(store, registered_models_names, rm_data):
+    model_name = registered_models_names[random_int(0, len(registered_models_names) - 1)]
```
Comment (BenWilson2) [inline tests/store/model_registry/test_file_store.py:139]: Why the random sampling here?

File: tests/store/model_registry/test_file_store.py [hunk 9]
```diff
@@ -1,1364 +1,1341 @@
...
+        source="A/C",
+        run_id=run_id_2,
+        tags=[ModelVersionTag("t1", "abc"), ModelVersionTag("t2", "x123")],
+    )
+    assert mv2.version == 2
+
+    def search_versions(filter_string):
+        return [mvd.version for mvd in store.search_model_versions(filter_string)]
+
+    assert search_versions(f"name = '{name}' and tag.t2 = 'xyz'") == [1]
```
Comment (BenWilson2) [inline tests/store/model_registry/test_file_store.py:821]: Can this be parameterized for this function?

File: tests/store/model_registry/test_file_store.py (hunk 2 above)
Comment (harupy) [reply tests/store/model_registry/test_file_store.py:34]: No, we can remove it.

Comment (wkrt7) [reply tests/store/model_registry/test_file_store.py]: deleted all scopes, resolved in https://github.com/wkrt7/mlflow/commit/bffd29e0461cf99eeac69de975a44c6b8361ace9

File: tests/store/model_registry/test_file_store.py (hunk 8 above)
Comment (wkrt7) [reply tests/store/model_registry/test_file_store.py:139]: I don't know. I kept it from the previous version.

File: tests/store/model_registry/test_file_store.py (hunk 7 above)
Comment (wkrt7) [reply tests/store/model_registry/test_file_store.py]: moved to the begining, https://github.com/wkrt7/mlflow/commit/7460619d7d5762a055d0d9ebb309c873ca9bdb2a

File: tests/store/model_registry/test_file_store.py (hunk 8 above)
Comment (BenWilson2) [reply tests/store/model_registry/test_file_store.py:139]: Yep, please simplify!

File: tests/store/model_registry/test_file_store.py (hunk 3 above)
Comment (harupy) [reply tests/store/model_registry/test_file_store.py:102]: I see, the original code contains this. Let's ignore it in this PR :)

**Comment by harupy** [comment]
@wkrt7 Can you resolve conflicts?

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
The goal of this PR is to migrate to pytest. We can improve tests in a follow-up PR.

**Pull request merged**